include/triton/core/tritonserver.h

```diff
@@ -1088,6 +1196,23 @@ TRITONSERVER_InferenceRequestAppendInputDataWithHostPolicy(
     const void* base, size_t byte_size, TRITONSERVER_MemoryType memory_type,
     int64_t memory_type_id, const char* host_policy_name);
 
+/// Assign a buffer of data to an input. The buffer will be appended
+/// to any existing buffers for that input. The 'inference_request'
+/// object takes ownership of the buffer and so the caller should not
```

Comment:
I updated this but removed it later. Things got a bit complicated if we wanted to transfer the ownership. In order to simplify things each of the buffers (output, input, state) have an internal buffer attributes object. If the frontends want to change it they will provide their own buffer attribute and it will just copy them into the internal buffer and the frontends should delete it separately.

Reply:
I see, so when this function get called, the content of `buffer_attributes` will be copied to the internal object.

Reply:
Correct.